flask_app/blueprints/rest.py

```diff
@@ -98,12 +98,25 @@ class TestResource(ModelResource):
     def _get_object_by_id(self, object_id):
         return _get_object_by_id_or_logical_id(self.MODEL, object_id)
 
+    def _render_many(self, objects, *, in_collection: bool):
+        rendered_tests = super()._render_many(objects, in_collection=in_collection)
+        if not rendered_tests or not current_user.is_authenticated:
+            return rendered_tests
+        if not in_collection:
+            rendered_tests['is_starred'] = db.session.query(db.session.query(UserStarredTests).filter(UserStarredTests.user_id == current_user.id, UserStarredTests.test_id==rendered_tests['id']).exists()).scalar()
+        else:
+            starred_test_ids = [test_id for (test_id, ) in db.session.query(Test.id).join(UserStarredTests).filter(UserStarredTests.user_id == current_user.id).all()]
```

Comment:
Make this a set for faster lookups

Reply:
You have to filter by test_id.in_(rendered_test_ids), because the user might have LOTS of starred tests, thus slowing down the query.